### PR TITLE
Reduce the memory consumption of TPC-C benchmark.

### PR DESCRIPTION
**Pull request by psudheer21**
Summary:
The original code was pre allocating a large array for the latency
results assuming that they were not going to use a lot of threads.
However, in the TPCC case we have a large number with each thread doing
very small number of operations. So this change reduces the size of the
arrays.

Removed some other components like histograms that were not being used
that contribute to the memory bloat.

Added some more cleanup of the code as well.

Reviewers:
Neha

### CODE REVIEW

File: src/com/oltpbenchmark/api/Worker.java [hunk 1]
```diff
@@ -57,17 +55,9 @@
...
-    protected final Map<Class<? extends Procedure>, Procedure> class_procedures = new HashMap<Class<? extends Procedure>, Procedure>();
-
-    private final Histogram<TransactionType> txnSuccess = new Histogram<TransactionType>();
-    private final Histogram<TransactionType> txnAbort = new Histogram<TransactionType>();
-    private final Histogram<TransactionType> txnRetry = new Histogram<TransactionType>();
-    private final Histogram<TransactionType> txnErrors = new Histogram<TransactionType>();
-    private final Map<TransactionType, Histogram<String>> txnAbortMessages = new HashMap<TransactionType, Histogram<String>>();
+    protected static WorkloadConfiguration wrkld;
+    protected  TransactionTypes transactionTypes;
+    protected  Map<Class<? extends Procedure>, Procedure> class_procedures = new HashMap<Class<? extends Procedure>, Procedure>();
```
Comment (ndeodhar) [inline src/com/oltpbenchmark/api/Worker.java]: Please extra space:
```suggestion
    protected TransactionTypes transactionTypes;
    protected Map<Class<? extends Procedure>, Procedure> class_procedures = new HashMap<Class<? extends Procedure>, Procedure>();
```

File: src/com/oltpbenchmark/api/Worker.java [hunk 2]
```diff
@@ -149,40 +137,20 @@ public final int getAndResetIntervalRequests() {
         return latencies;
     }
 
-    public final Procedure getProcedure(TransactionType type) {
-        return (this.procedures.get(type));
-    }
+    //public final Procedure getProcedure(TransactionType type) {
```
Comment (ndeodhar) [inline src/com/oltpbenchmark/api/Worker.java]: Can we remove instead of commenting out?

File: src/com/oltpbenchmark/api/Worker.java [hunk 3]
```diff
@@ -497,7 +454,7 @@ protected final TransactionType doWork(boolean measure, SubmittedProcedure piece
                                                ex.getClass().getSimpleName(), next, this.toString(),
                                                ex.getMessage(), ex.getErrorCode(), ex.getSQLState()), ex);
 
-                    this.txnErrors.put(next);
+                    //this.txnErrors.put(next);
```
Comment (ndeodhar) [inline src/com/oltpbenchmark/api/Worker.java]: Please remove

File: src/com/oltpbenchmark/api/Worker.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: Done.

File: src/com/oltpbenchmark/api/Worker.java (hunk 2 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: Done.

File: src/com/oltpbenchmark/api/Worker.java (hunk 3 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: Done.

**Review by ndeodhar — APPROVED** [review]

**Pull request merged**